core/templates/dev/head/pages/exploration_editor/ExplorationDataService.js

```diff
@@ -44,8 +44,15 @@ oppia.factory('ExplorationDataService', [
 
     var resolvedAnswersUrlPrefix = (
       '/createhandler/resolved_answers/' + explorationId);
-    var explorationDraftAutosaveUrl = (
-      '/createhandler/autosave_draft/' + explorationId);
+    var explorationDraftAutosaveUrl = '';
+    if (GLOBALS.can_edit) {
+      explorationDraftAutosaveUrl = (
+        '/createhandler/autosave_draft/' + explorationId);
+    } else {
```

Comment:
I think it may be better to do `else if (GLOBALS.can_translate)` and then in the final else clause put a warning of some sort in alerts service. This just makes it more obvious for future changes of this sort.

Reply:
Done!